packages/react-tabs/src/components/TabList/useTabList.ts

```diff
@@ -31,67 +31,33 @@ export const useTabList_unstable = (props: TabListProps, ref: React.Ref<HTMLElem
     initialState: undefined,
   });
 
+  // considered usePrevious, but it is sensitive to re-renders
+  // this could cause the previous to move to current in the case where the tab list re-renders.
+  const [previousSelectedValue, setPreviousSelectedValue] = useControllableState<TabValue | undefined>({
+    state: undefined,
+    defaultState: undefined,
+    initialState: undefined,
+  });
```

Comment:
Looks that this should just use `React.useState()` as this value cannot be controlled by consumers.

Reply:
does it even necessarily need to be a state ? from the way it is used it can also just be a ref

```ts
const previousSelectedValueRef = React.useRef();
setSelectedValue(selectedValue => {
  previousSelectedValueRef.current = selectedValue;
  return selectedValue;
})
```

Reply:
I had previously be using usePrevious, but it was sensitive to renders that caused current === previous incorrectly.  I can update to use a ref.